A sum type with a generic struct raises an error

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relevant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
**V version:** 0.2.4 a0e7a46
**OS:** linux, Linux version 5.13.0-39-generic (buildd@lcy02-amd64-080) (gcc (Ubuntu 9.4.0-1ubuntu1~~20.04.1) 9.4.0, GNU ld (GNU Binutils for Ubuntu) 2.34) #44~20.04.1-Ubuntu SMP Thu Mar 24 16:43:35 UTC 2022

<!-- Please include the standalone minimal reproducible code.
If not, it may be closed by the administrators. -->
**What did you do?**

Ran this script

```v
struct Just<T> {
  value T
}

struct Nothing {}

type Maybe = Nothing | Just
```

**What did you expect to see?**

No errors

 
**What did you see instead?**

```
/tmp/v_0/index.8598377962763286470.tmp.c:886:3: error: unknown type name 'main__Just_T_T'
  886 |   main__Just_T_T* _main__Just;
      |   ^~~~~~~~~~~~~~
```

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -492,7 +492,13 @@ pub fn (mut c Checker) sum_type_decl(node ast.SumTypeDecl) {
 			c.error('sum type cannot hold an interface', variant.pos)
 		} else if sym.kind == .struct_ && sym.language == .js {
 			c.error('sum type cannot hold an JS struct', variant.pos)
+		} else if mut sym.info is ast.Struct {
+			if sym.info.is_generic && !variant.typ.has_flag(.generic) {
+				c.error('generic struct `$sym.name` must specify generic type names, e.g. Foo<T>',
+					variant.pos)
+			}
 		}
+
 		if sym.name.trim_string_left(sym.mod + '.') == node.name {
 			c.error('sum type cannot hold itself', variant.pos)
 		}

--- a/vlib/v/checker/tests/generic_sumtype_decl_err.out
+++ b/vlib/v/checker/tests/generic_sumtype_decl_err.out
@@ -1,0 +1,7 @@
+vlib/v/checker/tests/generic_sumtype_decl_err.vv:7:24: error: generic struct `Just` must specify generic type names, e.g. Foo<T>
+    5 | struct Nothing {}
+    6 |
+    7 | type Maybe = Nothing | Just
+      |                        ~~~~
+    8 |
+    9 | fn main() {

--- a/vlib/v/checker/tests/generic_sumtype_decl_err.vv
+++ b/vlib/v/checker/tests/generic_sumtype_decl_err.vv
@@ -1,0 +1,10 @@
+struct Just<T> {
+  value T
+}
+
+struct Nothing {}
+
+type Maybe = Nothing | Just
+
+fn main() {
+}

